app/lib/features/space/pages/shell_page.dart

```diff
@@ -30,162 +19,12 @@ class SpaceShell extends ConsumerStatefulWidget {
-              Routes.spaceSettings.name,
-              pathParameters: {'spaceId': spaceId},
-            ),
-            child: const Text('Settings'),
-          ),
-        );
-      }
-    }
-
-    if (submenu.isNotEmpty) {
-      // add divider
-      submenu.add(const PopupMenuDivider());
-    }
-    submenu.add(
-      PopupMenuItem(
-        onTap: () => _handleLeaveSpace(context, space, ref),
-        child: const Text('Leave Space'),
-      ),
-    );
-
-    return Padding(
-      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 8),
-      child: Row(
-        children: [
-          InkWell(
-            onTap: () => context.canPop()
-                ? context.pop()
-                : context.goNamed(Routes.dashboard.name),
-            child: Icon(
-              Atlas.arrow_left,
-              color: Theme.of(context).colorScheme.neutral5,
-            ),
-          ),
-          const Spacer(),
-          Padding(
-            padding: const EdgeInsets.only(top: 18),
-            child: PopupMenuButton(
-              icon: Icon(
-                Icons.more_vert,
-                color: Theme.of(context).colorScheme.neutral5,
-              ),
-              iconSize: 28,
-              color: Theme.of(context).colorScheme.surface,
-              itemBuilder: (BuildContext context) => submenu,
-            ),
-          ),
-        ],
-      ),
-    );
-  }
-
-  void _handleLeaveSpace(
-    BuildContext context,
-    Space space,
-    WidgetRef ref,
-  ) {
-    showAdaptiveDialog(
-      barrierDismissible: false,
-      context: context,
-      builder: (context) => DefaultDialog(
-        title: Column(
-          children: <Widget>[
-            const Icon(Icons.person_remove_outlined),
-            const SizedBox(height: 5),
-            Text('Leave Space', style: Theme.of(context).textTheme.titleMedium),
-          ],
-        ),
-        subtitle: const Text(
-          'Are you sure you want to leave this space?',
-        ),
-        actions: <Widget>[
-          DefaultButton(
-            onPressed: () => context.pop(),
-            title: 'No Stay',
-            style: OutlinedButton.styleFrom(
-              side: BorderSide(color: Theme.of(context).colorScheme.success),
-            ),
-          ),
-          DefaultButton(
-            onPressed: () async {
-              await space.leave();
-              // refresh spaces list
-              ref.invalidate(spacesProvider);
-              if (!context.mounted) {
-                return;
-              }
-              context.pop();
-              context.pushNamed(Routes.dashboard.name);
-            },
-            title: 'Yes, Leave',
-            style: ElevatedButton.styleFrom(
-              backgroundColor: Theme.of(context).colorScheme.errorContainer,
-            ),
-          ),
-        ],
-      ),
+    return Column(
+      children: [
+        SpaceToolbar(spaceId: widget.spaceIdOrAlias),
+        ShellHeader(widget.spaceIdOrAlias),
```

Comment:
probably a similar renaming for the `ShellHeader`, too ... maybe `SpaceHeaderProfile`?